certificate-rotation.md

```diff
@@ -0,0 +1,113 @@
+There are two parts to certificate rotation: generating the new certificates, and reloading them.
+
+Both have been improved, with the main difference being that we can now reload certificates without restarting a node.
+
+## Order of certificate regeneration
+
+Most of the complication in rotating certificates is proper handling of new CA certificates.
+It's important for all parts of the system (nodes and clients) to have the same CA certificates, or they may not be able to verify each other's client/node certificates.
+
+It is possible to have multiple CA certificates active at the same time, but not node or client certificates. Therefore, CA certificates should be generated and pushed first, followed by new node/client certificates using the new CA.
+
+Let's say we want to regenerate all certificates (CA, client, node), we should do the following:
+1. generate new CA certificate and combine it with the existing one
+1. push combined CA certificate to all nodes and reload certificates
+1. make combined CA available to all clients. We may need to wait some time (days/weeks) for all clients to make use of the new certificate
```

Comment:
When we generate new node certificates (eg: adding a new node, or renewing the certificate on an old node), they will be signed by the new CA. This means that the CA needs to be on clients so that they can validate the new node cert.